crates/sui-framework/packages/deepbook/sources/clob_v2.move

```diff
@@ -839,19 +843,21 @@ module deepbook::clob_v2 {
                         filled_quote_quantity - filled_quote_quantity_without_commission,
                         maker_rebate
                     );
-                    vector::push_back(
-                        &mut matched_order_metadata, 
-                        matched_order_metadata(
-                            *object::uid_as_inner(&pool.id),
-                            account_owner(account_cap),
-                            maker_order,
-                            filled_base_quantity,
-                            // taker_commission = filled_quote_quantity - filled_quote_quantity_without_commission
-                            // This guarantees that the subtraction will not underflow
-                            filled_quote_quantity - filled_quote_quantity_without_commission,
-                            maker_rebate
-                        )
-                    );
+                    if(compute_metadata) {
```

Comment:
nice! space after `if` please.
Also the question of how big can the vector get stands